transport/http/outbound_test.go

```diff
@@ -683,6 +683,47 @@ func TestServiceMatchSuccess(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestGRPCInHTTPOut(t *testing.T) {
```

Comment:
(follow-up integration test idea) Should we have an integration test which verifies a gRPC request being forwarded to a HTTP outbound (and probably should do that for all combinations actually). Probably doesn't belong in transport/http but at the top-level probably

Reply:
Agree with this, I will take care of this in another PR though (less simple to review). The idea would be to have tests where request are being forwarded in a transport middleware where outbound transport will be different. 

This test actually checks how headers are actually updated or removed. I will update the name of the test though.